xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/AbstractClassPropertyValuesProvider.java

```diff
@@ -92,6 +98,31 @@ public PropertyValues getValues(ClassPropertyReference propertyReference, int li
         }
     }
 
+    @Override
+    public PropertyValues getValue(ClassPropertyReference propertyReference, Object... filterParameters)
+        throws XWikiRestException
+    {
+        String filter = "";
+        if (filterParameters.length > 0 && filterParameters[0] != null) {
+            filter = filterParameters[0].toString();
+        }
+        if (StringUtils.isEmpty(filter)) {
+            return new PropertyValues();
+        }
+
+        PropertyValues propertyValues = new PropertyValues();
+        DocumentReference documentReference = documentReferenceResolver.resolve(filter, EntityType.DOCUMENT);
+        if (documentReference != null) {
+            PropertyValue value =
+                this.getValueFromQueryResult(documentReference, getPropertyDefinition(propertyReference));
+            if (value != null) {
+                value.setValue(filter);
+                propertyValues.getPropertyValues().add(value);
+            }
+        }
+        return propertyValues;
```

Comment:
I would probably simply return:

    getValueFromQueryResult(rawValue, getPropertyDefinition(propertyReference))

and let derived classes overwrite this method if they want to resolve the raw value before calling ``getValueFromQueryResult``.